src/index.js

```diff
@@ -266,32 +251,25 @@ function recognizeChar(states, ch) {
   return nextStates;
 }
 
-var oCreate = Object.create || function(proto) {
-  function F() {}
-  F.prototype = proto;
-  return new F();
-};
-
-function RecognizeResults(queryParams) {
-  this.queryParams = queryParams || {};
+class RecognizeResults {
+  constructor(queryParams) {
+    this.splice = Array.prototype.splice;
+    this.slice = Array.prototype.slice;
+    this.push = Array.prototype.push;
+    this.length = 0;
```

Comment:
I think an array property on this class would be a cleaner API than the array-like result here, but need to figure out what that would affect.